v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go

```diff
@@ -211,3 +235,27 @@ func makeResourceReferenceProperty(idFactory astmodel.IdentifierFactory, existin
 
 	return newProp
 }
+
+func stripRemainingARMIDPrimitiveTypes(types astmodel.TypeDefinitionSet) (astmodel.TypeDefinitionSet, error) {
+	// Any astmodel.ARMReference's which have escaped need to be turned into
+	// strings
+	armIDStrippingVisitor := astmodel.TypeVisitorBuilder{
+		VisitPrimitive: func(_ *astmodel.TypeVisitor, it *astmodel.PrimitiveType, ctx interface{}) (astmodel.Type, error) {
+			if it == astmodel.ARMIDType {
```

Comment:
This is probably safe, but other type comparisons use the `TypeEquals()` method, so this should as well.

Reply:
Fixed